src/extension_instance.ml

```diff
@@ -434,39 +504,51 @@ let update_ocaml_info t =
   match ocaml_version with
   | Ok ocaml_version -> t.ocaml_version <- Some ocaml_version
   | Error e ->
-    (* [t.ocaml_version <- None] because we don't want [t.ocaml_version] to be
-       left over from a previous sandbox, which successfully set it *)
-    t.ocaml_version <- None;
-    (match e with
-     | `Unable_to_parse_version (`Version v, `Msg msg) ->
-       show_message
-         `Error
-         "OCaml bytecode compiler `ocamlc` version could not be parsed. Version: %s. \
-          Error %s"
-         v
-         msg
-     | `Ocamlc_missing ->
-       let (_ : unit Promise.t) =
-         let+ maybe_choice =
-           Window.showWarningMessage
-             ~message:
-               "OCaml bytecode compiler `ocamlc` was not found in the current sandbox. \
-                Do you have OCaml installed in the current sandbox?"
-             ~choices:
-               [ ( "Pick another sandbox"
-                 , fun () ->
-                     let (_ : Ojs.t option Promise.t) =
-                       Vscode.Commands.executeCommand
-                         ~command:Extension_consts.Commands.select_sandbox
-                         ~args:[]
-                     in
-                     () )
-               ]
-             ()
-         in
-         Option.iter maybe_choice ~f:(fun f -> f ())
+    (match t.sandbox with
+     | Dune _ ->
+       let open Promise.Option.Syntax in
+       let _ =
+         let+ project_root = Sandbox.workspace_root () |> Promise.return in
+         let* is_dune_pkg_locked = Sandbox.detect_dune_lock_dir ~project_root () in
+         match is_dune_pkg_locked with
+         | Dune _ -> Some () |> Promise.return
+         | _ -> Some (suggest_to_run_dune_pkg_lock t) |> Promise.return
```

Comment:
Suggesting to run dune pkg lock here won't actually fix the fact that the plugin wasn't able to detect OCaml's version, right ? Wouldn't it be more correct to do nothing here ? And later, when we will have a way to probe OCaml's version via Dune package management, then we will actually fix the check for real.